### PR TITLE
Schema handler fixes

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Updates and a minor bug fix to custom field saving.

Before
----------------------------------------
Unused function hanging around.
Wrong param passed to `fixSchemaDifferencesFor()`.
Overly complicated optimization that may have actually been less efficient.

After
----------------------------------------
Cleaner.

### CODE REVIEW

File: CRM/Core/BAO/CustomField.php [hunk 1]
```diff
@@ -145,7 +137,7 @@ public static function bulkSave($bulkParams, $defaults = []) {
         $logging->fixSchemaDifferencesFor($tableName, ['ADD' => $addedColumns[$tableName]]);
       }
 
-      Civi::service('sql_triggers')->rebuild($params['table_name'], TRUE);
+      Civi::service('sql_triggers')->rebuild($tableName, TRUE);
```
Comment (colemanw) [inline CRM/Core/BAO/CustomField.php:140]: @eileenmcnaughton this looks like it was a real bug to me.

File: CRM/Core/BAO/SchemaHandler.php [hunk 2]
```diff
@@ -271,43 +271,6 @@ public static function buildForeignKeySQL($params, $separator, $prefix, $tableNa
...
-   * @deprecated
-   *
-   * @param array $params
-   * @param bool $indexExist
-   * @param bool $triggerRebuild
-   *
-   * @return bool
-   */
-  public static function alterFieldSQL($params, $indexExist = FALSE, $triggerRebuild = TRUE) {
-    CRM_Core_Error::deprecatedFunctionWarning('function no longer in use / supported');
```
Comment (eileenmcnaughton) [inline CRM/Core/BAO/SchemaHandler.php:284]: Yep - deprecated Jul 2019

File: CRM/Core/BAO/CustomField.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Core/BAO/CustomField.php:140]: yes - it would have only updated one - probably in practice that was enough

**Comment by eileenmcnaughton** [comment]
Cover in testBulkCreate() is adequate here & code makes sense so as long as tests pass all good

**Pull request merged**